index.js

```diff
@@ -44,77 +44,182 @@ const redisClient = redis.createClient({
     try {
-      const shoe = await redisClient.lRange(owner, 0, -1);
-      if (shoe.length === 0) {
-        const owners = await redisClient.keys("*");
-        if (owners.includes(owner)) {
-          return res.send([{ owner: owner, shoes: [] }]);
-        }
-        throw new Error("Owner not found");
-      }
-      const listshoe = shoe.map((shoe) => JSON.parse(shoe));
-      const ownerShoes = { owner: owner, shoes: listshoe };
-      return res.send(ownerShoes);
+      const shoes = await redisClient.ft.search("idx:shoe", `@owner:${owner}`);
+      const shoesArray = shoes.documents.map((shoe) => {
+        const refactID = shoe.id.split(":");
+        return {
+          id: refactID[1],
+          color: shoe.value.color,
+          brand: shoe.value.brand,
+        };
+      });
+      return res.status(200).json({ owner: owner, shoes: shoesArray });
     } catch (error) {
       return res.status(404).send(error.message);
     }
   }
-  const owners = await redisClient.keys("*");
-  const shoes = [];
+  const owners = await redisClient.SMEMBERS("owners");
+  const jsonArray = [];
   for (const owner of owners) {
-    const shoe = await redisClient.lRange(owner, 0, -1);
-    const listshoe = shoe.map((shoe) => JSON.parse(shoe));
-    const ownerShoes = { owner: owner, shoes: listshoe };
-    shoes.push(ownerShoes);
+    try {
+      const shoes = await redisClient.ft.search("idx:shoe", `@owner:${owner}`);
+      const shoesArray = shoes.documents.map((shoe) => {
+        const refactID = shoe.id.split(":");
+        return {
+          id: refactID[1],
+          color: shoe.value.color,
+          brand: shoe.value.brand,
+        };
+      });
+      jsonArray.push({ owner: owner, shoes: shoesArray });
+    } catch (error) {
+      return res.status(404).send(error.message);
+    }
   }
-  res.send(shoes);
-});
-
-app.post("/dev-shoes", async (req, res) => {
-  const owners = ["John", "Jane", "Doe"];
-  const colors = ["red", "blue", "green"];
-  const brands = ["nike", "adidas", "reebok"];
-  const shoe = {
-    id: Math.floor(Math.random() * 1000),
-    brand: brands[Math.floor(Math.random() * brands.length)],
-    color: colors[Math.floor(Math.random() * colors.length)],
-  };
-  const owner = owners[Math.floor(Math.random() * owners.length)];
-  await redisClient.rPush(owner, JSON.stringify(shoe));
-  return res.json("Shoe added to Redis");
+  return res.status(200).json(jsonArray);
 });
 
+// Add a shoe to Redis Endpoint
 app.post("/shoes", async (req, res) => {
   const query = req.query;
   if (query.owner) {
     const owner = query.owner;
-    const body = req.body;
-    if (body.brand && body.color) {
-      body.id = Math.floor(Math.random() * 1000);
-      await redisClient.rPush(owner, JSON.stringify(body));
-      return res.status(200).send("Shoe added to Redis");
+    const color = req.body.color;
+    const brand = req.body.brand;
+    if (brand && color) {
+      const id = Math.floor(Math.random() * 100000);
```

Comment:
not sure if this might help, but you can use the redis `incr` function to make a non-random id.